plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php

```diff
@@ -132,6 +153,52 @@ public function register_routes() {
 		);
 	}
 
+	public function get_shipping_methods( $request ) {
+		$order = $this->get_object( (int) $request['id'] );
+
+		if ( ! $order || 0 === $order->get_id() ) {
+			return new WP_Error( "woocommerce_rest_{$this->post_type}_invalid_id", __( 'Invalid ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		$cart = new WC_Cart();
+		foreach ( $order->get_items( 'line_item' ) as $item ) {
+			$cart->add_to_cart( $item->get_product_id(), $item->get_quantity(), $item->get_variation_id() );
+		}
+
+		$packages = $cart->get_shipping_packages();
```

Comment:
Note that this may potentially change the cart items. If this is supposed to be used by admin users then it should be fine, but we should empty the cart after our calculations are done.

Some other alternatives to not have this method interfere with cart items would be:
1. Writing a corresponding `get_shipping_packages` in `WC_Order` class as well. Maybe we can extract out this method in a trait and then use the same code in both places.

2. We can also look into saving cart contents in a temporary variable before adding our order items, and then restore it back once our calculations are done.

Both of these alternatives are high effort, so please take a call on whether we would want to do these or not based on the use case.

Reply:
In this case the cart only exists for the current request and will be reconstructed from the order every time, so I think it should be OK? In general, I think 1 is the best long term solution. The REST API should be stateless and it's somewhat unexpected that we're using a cart in this way.